src/Sparrow/Platform/Posix/Syscall.cs

```diff
@@ -256,6 +257,44 @@ public static long lseek64(int fd, long offset, WhenceFlags whence)
         [DllImport(LIBC_6, SetLastError = true)]
         public static extern int mprotect(IntPtr start, ulong size, ProtFlag protFlag);
 
+        public static void PwriteOrThrow(int fd, byte *buffer, ulong count, long offset, string file, string debug)
+        {
+            Errno err = Errno.NONE;
+            bool cifsRetryOccured = false;
+            var actuallyWritten = 0L;
+            do
+            {
+                for (var cifsRetries = 0; cifsRetries < 3; cifsRetries++)
+                {
+                    var result = pwrite(fd, buffer, count - (ulong)actuallyWritten, offset + actuallyWritten);
+                    if (result < 0) // we assume zero cannot be returned at any case as defined in POSIX
+                    {
+                        err = (Errno)Marshal.GetLastWin32Error();
+                        if (err == Errno.EINVAL && CheckSyncDirectoryAllowed(file) == false)
```

Comment:
What about `EINTR`?

Reply:
EINTR is being handled here : https://github.com/ravendb/ravendb/pull/7620/files/1cb4e93e1d54b066253835b1bb6f1c6344f43b6a#diff-88eebaae7a92649e943992aa772ff384R323

Reply:
םל